contracts/libraries/State.sol

```diff
@@ -90,6 +90,17 @@ struct VaultAmmLiquidityPool {
     uint256 balance;
 }
 
+/**
+ * @dev LvInternalBalance structure for handling LV balances of users
+ */
+struct LvInternalBalance {
+    // must only be incremented when every user deposit to LV
+    uint256 balance;
+    // the value must not exceed the current balance.
+    // increment equal to the amount of LV every user withdrawal
+    uint256 withdrawn;
```

Comment:
it seems my initial spec design is flawed and we can keep only the balance here
- decrement it when user withdraw the PA
- increment if user deposit to LV

Reply:
this also needs a flag of dsId -> bool so that they can only redeem once per expiry. use OZ bitmap